extension-requests/script.js

```diff
@@ -331,3 +387,340 @@ filterButton.addEventListener('click', (event) => {
 
 populateStatus();
 render();
+
+async function createExtensionCard(data) {
+  //Api calls
+  const userDataPromise = getUserDetails(data.assignee);
+  const taskDataPromise = getTaskDetails(data.taskId);
+
+  const [{ taskData }, userData] = await Promise.all([
+    taskDataPromise,
+    userDataPromise,
+  ]);
+
+  const userImage = userData?.picture?.url ?? DEFAULT_AVATAR;
+  const userFirstName = userData?.first_name ?? data.assignee;
+  const isDeadLineCrossed = Date.now() > secondsToMilliSeconds(data.oldEndsOn);
+
+  const extensionDays = dateDiff(
+    secondsToMilliSeconds(data.newEndsOn),
+    secondsToMilliSeconds(data.oldEndsOn),
+  );
+  const deadlineDays = dateDiff(
```

Comment:
deadlineDays function is supposed to return the number of days between the current date and the old deadline date. However, the dateDiff() function is being passed the current date in milliseconds, and the old deadline date in seconds. This will result in an incorrect value being returned for the number of days.

Reply:
```javascript
const deadlineDays = dateDiff(
    Date.now(),
    secondsToMilliSeconds(data.oldEndsOn),
    (d) => d + (isDeadLineCrossed ? ' ago' : ''),
  );
  
function secondsToMilliSeconds(seconds) {
  return seconds * 1000;
}
``` 
I am converting old date to milliseconds using the function `secondsToMilliSeconds`